lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -230,7 +230,7 @@ class ClassMetadataInfo implements ClassMetadata
      * hierarchy. If the entity is not part of a mapped inheritance hierarchy this is the same
      * as {@link $name}.
      *
-     * @var string
+     * @var class-string
```

Comment:
```suggestion
     * @var string
     * @psalm-var class-string
```

Reply:
What's the rationale for this? PHPDocumentor and PHPStorm both understand this notation now. Is there another tool we want compatibility with that doesn't support this?

Reply:
Someone with an older version of PHPStorm ? Someone with another IDE ?

Reply:
@greg0ire WDYT? Should we keep class-string in @psalm annotations? What's the policy for doctrine repos?

Reply:
For Doctrine repositories we use `@psalm` annotations historically, because of PHPStorm, but we might want to relax that, I hear that indeed PHPStorm understands these now.

Reply:
Thanks! Maybe that's something that could be enforced via CS rules, one way or another

Reply:
After discussing this, we are not yet ready to enforce it, but we do think we should no longer require contributors to add prefixes.

Reply:
Good to know, thanks!

I added the psalm prefix for now but I'll keep that in mind for next time :)

Reply:
Yup, also, as pointed out internally by @ostrolucky, there are some tricky cases like cases involving template that should stay with prefixes.

Reply:
yes. Also phpstorm doesn't yet read *all* annotations. Array shapes for example will be reported as an error, even more so with list (for example `list<string, array{string, string}>` will be a `undefined class list`)